### ISSUE TITLE
Flat Table with sorting headers has critical accessibility warnings

### ISSUE THREAD

**Issue by Solkarim91** [Sage/carbon#7972]
### Description

Getting critical warnings "ARIA attributes must conform to valid values"
It appears that the GUID generated for the sortable column headers `aria-describedby` attribute is not being recognised in the DOM

From Carbon docs:
<img width="1617" height="910" alt="Image" src="https://github.com/user-attachments/assets/595a878f-8742-4f43-8fbd-8a3d377e4b51" />

Failing autotests from SBCP:
<img width="3380" height="1484" alt="Image" src="https://github.com/user-attachments/assets/0afa3361-eaf7-4ab5-abdd-493e4ed8b956" />

### Reproduction

https://carbon.sage.com/?path=/story/flat-table--with-sorting-headers&args=;;

### JIRA ticket numbers (Sage only)

SKPU-16081

### Carbon version

158.41.2

### React version

v17

### Design tokens version

4.35.0

### Relevant browsers

Chrome

### Relevant OSs

MacOS

### Confidentiality

- [x] I confirm there is no confidential or commercially sensitive information included.

**Comment by nuria1110** [comment]
FE-7685

**Comment by jamessandom** [comment]
This issue is also present in the employee table in payroll and causes accessibility automation scripts to fail

<img width="1832" height="972" alt="Image" src="https://github.com/user-attachments/assets/6825b3e4-50a7-46d9-a8e0-7b3349a84203" />

also

<img width="1585" height="676" alt="Image" src="https://github.com/user-attachments/assets/e1746d83-3be0-49c4-b2c4-1f3f06628136" />

**Comment by jamessandom** [comment]
The below Accessibility tests are affected by this issue:

UK

EmployeeBulkActionsBarAccessibilityTest
EmployeeCustomiseDialogAccessibilityTest
AddedEmployeeAccessibilityTest
EmployeeFilterDialogAccessibilityTest
EmployeesLandingPageAccessibilityTest
NewEmployeeDialogAccessibilityTest
PayRunProSidePanelAbsenceAccessibilityTest
PayRunProSidePanelPaymentPresentAccessibilityTest
PayRunProSidePanelDeductionPresentAccessibilityTest

ROI

AddPayslipMessageDialogAccessibilityTest
EmployeeBulkActionsBarAccessibilityTest
EmployeeCustomiseDialogAccessibilityTest
EmployeeFilterDialogAccessibilityTest
AddedEmployeeAccessibilityTest
EmployeesLandingPageAccessibilityTest
NewEmployeeDialogAccessibilityTest

**Comment by edleeks87** [comment]
We'll remove the id and `aria-describedby` as they aren't needed anymore and are causing the incomplete axe checks
